test/e2e/agent/agent_test.go

```diff
@@ -63,4 +64,317 @@ var _ = Describe("VM Agent behavior", func() {
+			harness.ApproveEnrollment(deviceId, testutil.TestEnrollmentApproval())
+			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
+				deviceId).ShouldNot(BeEmpty())
+			logrus.Infof("The device %s was approved", deviceId)
+
+			// Wait for the device to pickup enrollment and report measurements on device status.
+			Eventually(harness.GetDeviceWithStatusSystem, TIMEOUT, POLLING).WithArguments(
+				deviceId).ShouldNot(BeNil())
+			logrus.Infof("The device %s is reporting its status", deviceId)
+
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				logrus.Infof("Current device is %s", deviceId)
+
+				// Create ConfigProviderSpec.
+				var configProviderSpec v1alpha1.ConfigProviderSpec
+				err := configProviderSpec.FromGitConfigProviderSpec(gitConfigInvalidRepo)
+				Expect(err).ToNot(HaveOccurred())
+
+				device.Spec.Config = &[]v1alpha1.ConfigProviderSpec{configProviderSpec}
+				logrus.Infof("Updating %s with config %s", deviceId, device.Spec.Config)
+			})
+
+			// Check the http config error is detected.
+			harness.WaitForDeviceContents(deviceId, `1 invalid configuration: example-git-config-provider. Error: failed fetching
+			specified Repository definition 00000000-0000-0000-0000-000000000000/not-existing-repo:
+			resource not found`,
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "SpecValid", "False", "Invalid")
+				}, "2m")
+
+			// The behaviour will change after EDM-418.
+			harness.WaitForDeviceContents(deviceId, "the device is updated to renderedVersion: 1",
+				func(device *v1alpha1.Device) bool {
+					return conditionExists(device, "Updating", "False", "Updated")
+				}, "2m")
+			Eventually(harness.GetDeviceWithStatusSummary, TIMEOUT, POLLING).WithArguments(
+				deviceId).Should(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+		})
+
+		It(`should show an error when trying to update a device with a httpConfigProviderSpec
+			with invalid Path, and report 'Online' status`, func() {
+			deviceId := harness.GetEnrollmentIDFromConsole()
+
+			// Approve the enrollment and wait for the device details to be populated by the agent.
+			harness.ApproveEnrollment(deviceId, testutil.TestEnrollmentApproval())
+			Eventually(harness.GetDeviceWithStatusSummary, LONGTIMEOUT, POLLING).WithArguments(
+				deviceId).ShouldNot(BeEmpty())
+			logrus.Infof("The device %s was approved", deviceId)
+
+			// Wait for the device to pickup enrollment and report measurements on device status.
+			Eventually(harness.GetDeviceWithStatusSystem, TIMEOUT, POLLING).WithArguments(
+				deviceId).ShouldNot(BeNil())
+			logrus.Infof("The device %s is reporting its status", deviceId)
+
+			// Create the http repository.
+			out, err := harness.CLIWithStdin(httpRepo, "apply", "-f", "-")
+			Expect(err).ToNot(HaveOccurred())
+			Expect(out).To(MatchRegexp(`(200 OK|201 Created)`))
```

Comment:
I would suggest we also use the API h.Client.  to create a new repo instead, the advantage of this, as we discussed is that any change in the structures or format is caught by the developers when changing the API, so you repo content/format "yaml" gets also updated without your intervention.

Reply:
Done, thanks